karabo_bridge/simulation.py

```diff
@@ -62,33 +63,32 @@ def gen_combined_detector_data(detector_info, tid_counter, corrected=False, nsou
         'timestamp.frac': frac.ljust(18, '0')  # attosecond resolution
     }
 
-    if corrected and detector_info['source'] == 'SPB_DET_AGIPD1M-1/DET/detector':
-        # Corrected data from AGIPD might halve the number of frames and cut off
-        # the first one.
-        # TODO: Check if this is still true
-        pulse_count = 31
-    else:
-        pulse_count = detector_info['pulses']
-    array_shape = (pulse_count, detector_info['modules']) + detector_info['module_shape']
+    pulse_count = detector_info['pulses']
+    array_shape = tuple((detector_info['modules'], ) +
+                        detector_info['module_shape'] + (pulse_count, ))
 
     # detector random data
     if corrected:
         gain_data = np.zeros(array_shape, dtype=np.uint16)
+        domain = detector_info['source'].partition('/')[0]
         passport = [
-            'SPB_DET_AGIPD1M-1/CAL/THRESHOLDING_Q3M2',
-            'SPB_DET_AGIPD1M-1/CAL/OFFSET_CORR_Q3M2',
-            'SPB_DET_AGIPD1M-1/CAL/RELGAIN_CORR_Q3M2'
+            '%s/CAL/THRESHOLDING_Q1M1' % domain,
+            '%s/CAL/OFFSET_CORR_Q1M1' % domain,
+            '%s/CAL/RELGAIN_CORR_Q1M1' % domain
         ]
 
         gen[source]['image.gain'] = gain_data
         gen[source]['image.passport'] = passport
 
     rand_data = partial(np.random.uniform, low=1500, high=1600,
                         size=detector_info['module_shape'])
-    data = np.zeros(array_shape, dtype=np.uint16)  # np.float32)
+    if corrected:
+        data = np.zeros(array_shape, dtype= np.float32)
+    else:
+        data = np.zeros(array_shape, dtype=np.uint16)
     for pulse in range(pulse_count):
         for module in range(detector_info['modules']):
-            data[pulse, module, ] = rand_data()
+            data[module, :, :, pulse] = rand_data()
```

Comment:
I tried the other day creating the whole array with `np.random.uniform()`, but bizarrely this seemed to be slower than making an array of zeros and then filling slices with random data in a loop.  Do you know why that is?